pkg/provisioner/ironic/ironic.go

```diff
@@ -1377,6 +1408,38 @@ func (p *ironicProvisioner) Deprovision(force bool) (result provisioner.Result,
 		return transientError(provisioner.NeedsRegistration)
 	}
 
+	p.log.Info("WARNING: automated cleaning is disabled for host",
```

Comment:
Seems like we should refactor this into a function, given that the Provision/Deprovision code is the same?

Reply:
Same here . The log is coming before the check

Reply:
+1 for refactoring it into a function.